### ISSUE TITLE
Knit Directory setting can break R Markdown websites

### ISSUE THREAD

**Issue by jdblischak** [rstudio/rstudio#2158]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.1.423
    OS Version      : Ubuntu 17.10 (artful)
    R Version       : 3.4.3

### Steps to reproduce the problem

1. Create an empty file called `_site.yml`
1. Open a new R Markdown file and save it as `index.Rmd`  in the same directory as `_site.yml`
1. Set the Knit Directory to Document Directory
1. Click Knit (it works)
1. Set the Knit Directory to Current Directory
1. Click Knit (error)

### Describe the problem in detail

RStudio v1.1 introduced a new feature to [change the knit directory](https://blog.rstudio.com/2017/09/13/rstudio-v1.1---the-little-things) via PR #904. This new feature causes problems for [R Markdown websites](http://rmarkdown.rstudio.com/rmarkdown_websites.html). The argument `knit_root_dir` can be passed directly to `rmarkdown::render`, but RStudio uses `rmarkdown::render_site` if it [detects a website](https://github.com/rstudio/rstudio/blob/97f70cf20d31c7e8e7a38db65a3410040ce74388/src/cpp/session/modules/SessionRMarkdown.R#L102). Because `rmarkdown::render_site` does not have the argument `knit_root_dir`, it throws an error like the following:

```
Error in rmarkdown::render_site("/home/jdb-work/index.Rmd", encoding = "",  : 
  unused argument (knit_root_dir = "~")
Execution halted
```

### Describe the behavior you expected

Since `rmarkdown::render_site` does not have the argument `knit_root_dir`, I expected RStudio to ignore the Knit Directory setting when executing `rmarkdown::render_site`.

I am using [rmarkdown](https://cran.r-project.org/package=rmarkdown) 1.8, which is the latest CRAN release. Please let me know if there is any other information that would be useful.

**Comment by ronblum** [comment]
@jdblischak Thank you for notifying us of this issue!

**Comment by jjallaire** [comment]
@jmcphers If this really breaks render_site this might be a candidate for a patch release.

**Comment by jdblischak** [comment]
> However, when working in a project, these work.

@ronblum I cannot replicate this. Even in the context of an RStudio project, I still get the same error when either "Current Working Directory" or "Project Directory" is set. In both cases, the argument `knit_root_dir` is passed to `render_site`, which causes an error. Could you please provide the steps you took that allowed you to run `render_site` via RStudio without error?

My directory structure looked like this:

```
$ tree .
.
├── index.Rmd
├── _site.yml
└── test-render_site.Rproj

0 directories, 3 files
```

**Comment by ronblum** [comment]
Sorry! Ignore my previous comment—I left out a step when I created a project instead of no-project. It fails there, too.

**Comment by jdblischak** [comment]
A minor follow up. This will also affect the [Getting Started directions for bookdown](https://bookdown.org/home/getting-started.html#edit) since bookdown uses `rmarkdown::render_site`.

**Comment by jmcphers** [comment]
@jdblischak Thanks very much for reporting this! We missed this combination when setting up the Knit in Directory feature set. I've just pushed a fix, and will make the fix part of the next 1.1 release should we roll one. 

Hopefully it won't be likely in the wild as people are unlikely to try to swap the knit directory when working with websites!

**Comment by jdblischak** [comment]
> Thanks very much for reporting this! We missed this combination when setting up the Knit in Directory feature set. I've just pushed a fix, and will make the fix part of the next 1.1 release should we roll one.

Thanks for fixing this, @jmcphers! I look forward to the next release.

> Hopefully it won't be likely in the wild as people are unlikely to try to swap the knit directory when working with websites!

Interesting. What makes you say that? I'd assume the opposite. With just a few R Markdown files, it is easy enough to save them in the directory where you want them to be executed (i.e. no need to change the knit directory). On the other hand, with a website with many R Markdown files, it is cumbersome to save them all in the place to be executed, especially if you want the files to be executed from the project directory.

**Comment by mikoontz** [comment]
I just did this in the wild! (Broke my website by having "Evaluate chunks in directory: Project" set from another session and got a `unused argument (knit_root_dir = "~")` error). Switching the Knit Directory back to the Document Directory fixed the issue.

I was playing around with the "Evaluate chunks in directory" dropdown menu in RStudio -> Preferences -> R Markdown to see if it did what I thought it did, and didn't realize I could also change it under the "Knit" dropdown menu.

Not sure whether this is related, but the dropdown menu for "Evaluate chunks in directory" in the RStudio -> Preferences -> R Markdown doesn't seem to reflect the currently set Knit Directory. It always shows "Document" even if I switch it to "Project", hit "Okay", have the Rmd successfully knit using the Project directory as the knit directory, then go back to the RStudio -> Preferences -> RMarkdown tab. The checkbox under the Knit -> Knit Directory dropdown does move between "Project Directory" and "Document Directory". Is this expected behavior?
